crates/ide_assists/src/handlers/convert_iter_for_each_to_for.rs

```diff
@@ -0,0 +1,214 @@
+use ide_db::helpers::FamousDefs;
+use syntax::{
+    ast::{self, edit::AstNodeEdit, make, ArgListOwner},
+    AstNode,
+};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+/// Assist: convert_iter_for_each_to_for
+//
+/// Converts an Iterator::for_each function into a for loop.
+///
+/// ```rust
+/// fn main() {
+///     let vec = vec![(1, 2), (2, 3), (3, 4)];
+///     x.iter().for_each(|(x, y)| {
+///         println!("x: {}, y: {}", x, y);
+///    })
+/// }
+/// ```
+/// ->
+/// ```rust
+/// fn main() {
+///     let vec = vec![(1, 2), (2, 3), (3, 4)];
+///     for (x, y) in x.iter() {
+///         println!("x: {}, y: {}", x, y);
+///     });
+/// }
+/// ```
```

Comment:
```suggestion
// Assist: convert_iter_for_each_to_for
//
// Converts an Iterator::for_each function into a for loop.
//
// ```rust
// fn main() {
//     let vec = vec![(1, 2), (2, 3), (3, 4)];
//     x.iter().for_each(|(x, y)| {
//         println!("x: {}, y: {}", x, y);
//     });
// }
// ```
// ->
// ```rust
// fn main() {
//     let vec = vec![(1, 2), (2, 3), (3, 4)];
//     for (x, y) in x.iter() {
//         println!("x: {}, y: {}", x, y);
//     }
// }
// ```
```
We don't use doc-comments for these, also make sure this generates a test fixture which should happen by running `cargo test -p xtask`